scripts/speech_recognition/confidence/benchmark_asr_confidence.py

```diff
@@ -0,0 +1,427 @@
+        autocast = torch.cuda.amp.autocast
+    else:
+
+        @contextlib.contextmanager
+        def autocast():
+            yield
+
+    # do grig-based benchmarking if grid_params is provided, otherwise a regular one
+    work_dir = Path(cfg.output_dir)
+    os.makedirs(work_dir, exist_ok=True)
+    report_legend = (
+        ",".join(
+            [
+                "model_type",
+                "aggregation",
+                "blank",
+                "method_name",
+                "entropy_type",
+                "entropy_norm",
+                "temperature",
+                "target_level",
+                "auc_roc",
+                "auc_pr",
+                "auc_nt",
+                "nce",
+                "ece",
+                "auc_yc",
+                "max_yc",
+                "std_yc",
+            ]
+        )
+        + "\n"
+    )
+    model_typename = "RNNT" if is_rnnt else "CTC"
+    if cfg.grid_params:
+        report_file = work_dir / Path("report.csv")
+        asr_model.change_decoding_strategy(
+            RNNTDecodingConfig(fused_batch_size=-1, strategy="greedy_batch", confidence_cfg=cfg.confidence_cfg)
+            if is_rnnt
+            else CTCDecodingConfig(confidence_cfg=cfg.confidence_cfg)
+        )
+        params = json.loads(cfg.grid_params)
+        hp_grid = ParameterGrid(params)
+        hp_grid = list(hp_grid)
+
+        logging.info(f"==============================Running a benchmarking with grid search=========================")
+        logging.info(f"Grid search size: {len(hp_grid)}")
+        logging.info(f"Results will be written to:\nreport file `{report_file}`\nand plot directories near the file")
+        logging.info(f"==============================================================================================")
+
+        with open(work_dir / Path("report.csv"), "tw", encoding="utf-8") as f:
+            f.write(report_legend)
+            f.flush()
+            for i, hp in enumerate(hp_grid):
+                logging.info(f"Run # {i + 1}, grid: `{hp}`")
+                asr_model.change_decoding_strategy(apply_parameters(asr_model.cfg.decoding, hp))
+                param_list, experiment_name = get_experiment_params(asr_model.cfg.decoding.confidence_cfg)
+                plot_dir = work_dir / Path(experiment_name)
+                results = run_benchmark(
+                    asr_model,
+                    cfg.batch_size,
+                    cfg.num_workers,
+                    is_rnnt,
+                    cfg.target_level,
+                    filepaths,
+                    reference_texts,
+                    plot_dir,
+                    autocast,
+                )
+                for level, result in results.items():
+                    f.write(f"{model_typename},{','.join(param_list)},{level},{','.join([str(r) for r in result])}\n")
+                    f.flush()
+    else:
+        report_file = work_dir / Path("report.csv")
```

Comment:
Line 357 and line 395 duplicate each other. Please move that line out of if/else scope to the line before line 356.